### ISSUE TITLE
How could I use this within a project using React + Tailwind + Vite?

### ISSUE THREAD

**Issue by sort72** [n4j1Br4ch1D/postcss-obfuscator#2]
Hi, I like this plugin, but I'm not able to install it on my project which uses Tailwind, React, Vite and PostCSS, could you offer some guidance, please?

I saw the instructions in Astro + Tailwind example but it won't work for me

Thanks!

**Comment by sort72** [comment]
I used 'yarn build' and it generated the following output in console:

![Output](https://i.imgur.com/EE9WMyF.jpeg "output")

And it also generated 'out' and 'dist' directories, however if I access 'dist/index.html' file, it says:

Unexpected Application Error!
404 Not Found

**Comment by n4j1Br4ch1D** [comment]
Hi, Thanks for Asking.

### I got it running with React + Vite + TailwindCSS :
<img width="649" alt="postcss-obfuscator-react-vite-tailwindcss" src="https://user-images.githubusercontent.com/38995898/236050732-ee5b3cc3-ba3d-4beb-aedd-5f056d4ab634.png">

### Fork this Replit:
https://replit.com/@n4j1Br4ch1D/postcss-obfuscator-react-vite-tailwindcss

### How it's done:

1- install  postcss-obfuscator:
```sh
# npm
npm install postcss-obfuscator --save-dev
# yarn
yarn add postcss-obfuscator --dev
```
2- install tailwindcss  and configure it:
```js
# Using NPM
npm install -D tailwindcss postcss autoprefixer
npx tailwindcss init -p
```
3- in src/index.css import tailwindcss:
```css
@tailwind base;
@tailwind components;
@tailwind utilities;
```

4- create postcss.config.cjs  his minimum config I used in demo
```js
//postcss.config.cjs 

process.env.NODE_ENV = "development" //development //obsfucation //production
const isObfscMode = process.env.NODE_ENV === 'obsfucation'; 

module.exports = {
  plugins: [
    require('tailwindcss'),
    require('autoprefixer'),
    require('postcss-obfuscator')({
    /* options */
      enable: isObfscMode, // Force Run on Dev Env or when srcPath equals desPath.
      srcPath: "src", // Source of your files.
      desPath: "out", // Destination for obfuscated html/js/.. files.
      extensions: ['.jsx'],
      formatJson: true, // Format obfuscation data JSON file.
    })
  ]
}
```

5- optional: add this npm script's to package.json
```json
    "postcss": "postcss src/**/*.css  --dir out/css",
    "obfuscate": "npm run postcss"
```
6- Install postcss-cli:
```sh
npm i -D postcss postcss-cli
```
7- now to obfuscate,  set the env mode to `obsfucation` ,  and in your console run :
```sh
 npm run obfuscate 
```
which will generate obfuscated JSON file, with obfuscated files in an output folder.

__Note:__ postcss is automatically run with Vite so its preferred to create a customer postcss script to run only when you want to obfuscate ready for production see the readme for more details. and feel free to contact me again. 
Also you can use two tricks: 
1. the `enable` option (as used in demo) to enable it only in specific mode and to make sure tailwindcsss works fine in dev mode:
```js
process.env.NODE_ENV = "development" //development //obsfucation //production
const isObfscMode = process.env.NODE_ENV === 'obsfucation'; 

//enable: isObfscMode,
```
2. the `callBack` option Callback function to call after obfuscation is done.  that way once obfuscation is done you can config vite to read data from `out` folder instead of  `src` or you can just rename the folders.

**Comment by n4j1Br4ch1D** [comment]
> I used 'yarn build' and it generated the following output in console:
> 
> ![Output](https://camo.githubusercontent.com/5f5dfdb70330b82bdbd181d4d3d6755a2afc5be691b096827b221ebcd127a41b/68747470733a2f2f692e696d6775722e636f6d2f454539574d79462e6a706567)
> 
> And it also generated 'out' and 'dist' directories, however if I access 'dist/index.html' file, it says:
> 
> Unexpected Application Error! 404 Not Found

i think that a different issue for me it worked fine so plz try this:
in `dist/index.html`
change url paths (remove the slash/)
instead of `/assets/...`  should be `assets/...` and so on
```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <link rel="icon" type="image/svg+xml" href="vite.svg" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite + React</title>
    <script type="module" crossorigin src="assets/index-6e17a1d2.js"></script>
    <link rel="stylesheet" href="assets/index-fffba02c.css">
  </head>
  <body>
    <div id="root"></div>
    
  </body>
</html>
```
@sort72  plz if it the issue was resolved then close it. else contact me back I don't close issues until they truly solved and the issuer close it himself. and consider staring the project to keep the project going.

**Comment by sort72** [comment]
> Hi, Thanks for Asking.
> 
> ### I got it running with React + Vite + TailwindCSS :
> <img alt="postcss-obfuscator-react-vite-tailwindcss" width="649" src="https://user-images.githubusercontent.com/38995898/236050732-ee5b3cc3-ba3d-4beb-aedd-5f056d4ab634.png">
> 
> ### Fork this Replit:
> https://replit.com/@n4j1Br4ch1D/postcss-obfuscator-react-vite-tailwindcss
> 
> ### How it's done:
> 1- install postcss-obfuscator:
> 
> ```shell
> # npm
> npm install postcss-obfuscator --save-dev
> # yarn
> yarn add postcss-obfuscator --dev
> ```
> 
> 2- install tailwindcss and configure it:
> 
> ```js
> # Using NPM
> npm install -D tailwindcss postcss autoprefixer
> npx tailwindcss init -p
> ```
> 
> 3- create `src/input.css` and import tailwindcss:
> 
> ```css
> @tailwind base;
> @tailwind components;
> @tailwind utilities;
> ```
> 
> 4- create `postcss.config.cjs` this minimum config I used in demo
> 
> ```js
> //postcss.config.cjs 
> 
> process.env.NODE_ENV = "development" //development //obfuscation //production
> const isObfscMode = process.env.NODE_ENV === 'obfuscation'; 
> 
> module.exports = {
>   plugins: [
>     require('tailwindcss'),
>     require('autoprefixer'),
>     require('postcss-obfuscator')({
>     /* options */
>       enable: isObfscMode, // Force Run on Dev Env or when srcPath equals desPath.
>       srcPath: "src", // Source of your files.
>       desPath: "out", // Destination for obfuscated html/js/.. files.
>       extensions: ['.jsx'],
>       formatJson: true, // Format obfuscation data JSON file.
>     })
>   ]
> }
> ```
> 
> 5- add this npm script's to package.json
> 
> ```json
>     "tailwindcss": "npx tailwindcss -i ./src/input.css -o ./src/output.css",
>     "postcss": "postcss src/**/*.css  --dir out/css",
>     "obfuscate": "npm run tailwindcss && npm run postcss"
> ```
> 
> 6- Install postcss-cli:
> 
> ```shell
> npm i -D postcss postcss-cli
> ```
> 
> 7- now to obfuscate, set the env mode to `obfuscation` , and in your console run :
> 
> ```shell
>  npm run obfuscate 
> ```
> 
> which will generate obfuscated JSON file, with obfuscated files in an output folder.
> 
> **Note:** postcss is automatically run with Vite so its preferred to create a customer postcss script to run only when you want to obfuscate ready for production see the readme for more details. and feel free to contact me again. Also you can use two tricks:
> 
> 1. the `enable` option (as used in demo) to enable it only in specific mode and to make sure tailwindcsss works fine in dev mode:
> 
> ```js
> process.env.NODE_ENV = "development" //development //obfuscation //production
> const isObfscMode = process.env.NODE_ENV === 'obfuscation'; 
> 
> //enable: isObfscMode,
> ```
> 
> 2. the `callBack `option, a Callback function to call after obfuscation is done. that way once obfuscation is done you can config and prepare your project for production: so basically you use `callBack` option to set the env mode back to `production` so that obsfucation will not run, and then config your app source folder to use `out` folder instead of `src` for production. and the imported tawilwindcss file. if u want css file to be obfuscated to same path just set the ouput directory of postcss directly to `out`
> 
> ```
>  "postcss": "postcss src/**/*.css  --dir out",
> ```
> 
> so basically you use `callBack` option to set the env mode back to `production` and to source folder of production to `out` instead of `src` that way it will work. when you run `npm run build`

Hi, thanks for your response

I did the steps you mentioned and I'm almost there, but I have a problem I can't understand why is happening, it looks like a bug

I run first 

```shell
  yarn obfuscate
```

Then

```shell
 yarn build
```

And it's showing some warnings, because I was using classes with arbitrary values before (Like h-[0.5rem]) and the parser was not able to obfuscate them properly, showing those warnings.

Then I removed the arbitrary values and replaced them for built-in tailwind values, then obfuscated it again and all that. But, when I build it one more time, it still shows warnings (Even when I'm not using the arbitrary values anymore). If I search any of the warning classes, none of them exist! See the photo:

![Screenshot_10](https://user-images.githubusercontent.com/47725212/236218267-8a03b2b4-efab-41ce-b91b-1f37e44cffd4.jpg)

So I ignored warning and renamed out to src and built it, but it was complainig about a syntax error in the CSS. I fixed it and built again, with a successful built, then I opened my project but none of the styles were there. The css classes in the html tags were named different than in the CSS file (For example, in the html class, the name was 'abcd' and in the css file, that class had the 'xyz' name). 

Maybe this last issue is related with warnings thing, I removed css-obfuscator, dist and out folders before building again (I also removed node_modules, and restarted the pc, lol), and the warnings were still there.

Any hint? 

Thank you!

**Comment by n4j1Br4ch1D** [comment]
> > Hi, Thanks for Asking.
> > ### I got it running with React + Vite + TailwindCSS :
> > <img alt="postcss-obfuscator-react-vite-tailwindcss" width="649" src="https://user-images.githubusercontent.com/38995898/236050732-ee5b3cc3-ba3d-4beb-aedd-5f056d4ab634.png">
> > ### Fork this Replit:
> > https://replit.com/@n4j1Br4ch1D/postcss-obfuscator-react-vite-tailwindcss
> > ### How it's done:
> > 1- install postcss-obfuscator:
> > ```shell
> > # npm
> > npm install postcss-obfuscator --save-dev
> > # yarn
> > yarn add postcss-obfuscator --dev
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 2- install tailwindcss and configure it:
> > ```js
> > # Using NPM
> > npm install -D tailwindcss postcss autoprefixer
> > npx tailwindcss init -p
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 3- create `src/input.css` and import tailwindcss:
> > ```css
> > @tailwind base;
> > @tailwind components;
> > @tailwind utilities;
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 4- create `postcss.config.cjs` this minimum config I used in demo
> > ```js
> > //postcss.config.cjs 
> > 
> > process.env.NODE_ENV = "development" //development //obfuscation //production
> > const isObfscMode = process.env.NODE_ENV === 'obfuscation'; 
> > 
> > module.exports = {
> >   plugins: [
> >     require('tailwindcss'),
> >     require('autoprefixer'),
> >     require('postcss-obfuscator')({
> >     /* options */
> >       enable: isObfscMode, // Force Run on Dev Env or when srcPath equals desPath.
> >       srcPath: "src", // Source of your files.
> >       desPath: "out", // Destination for obfuscated html/js/.. files.
> >       extensions: ['.jsx'],
> >       formatJson: true, // Format obfuscation data JSON file.
> >     })
> >   ]
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 5- add this npm script's to package.json
> > ```json
> >     "tailwindcss": "npx tailwindcss -i ./src/input.css -o ./src/output.css",
> >     "postcss": "postcss src/**/*.css  --dir out/css",
> >     "obfuscate": "npm run tailwindcss && npm run postcss"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 6- Install postcss-cli:
> > ```shell
> > npm i -D postcss postcss-cli
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 7- now to obfuscate, set the env mode to `obfuscation` , and in your console run :
> > ```shell
> >  npm run obfuscate 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > which will generate obfuscated JSON file, with obfuscated files in an output folder.
> > **Note:** postcss is automatically run with Vite so its preferred to create a customer postcss script to run only when you want to obfuscate ready for production see the readme for more details. and feel free to contact me again. Also you can use two tricks:
> > 
> > 1. the `enable` option (as used in demo) to enable it only in specific mode and to make sure tailwindcsss works fine in dev mode:
> > 
> > ```js
> > process.env.NODE_ENV = "development" //development //obfuscation //production
> > const isObfscMode = process.env.NODE_ENV === 'obfuscation'; 
> > 
> > //enable: isObfscMode,
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > 2. the `callBack `option, a Callback function to call after obfuscation is done. that way once obfuscation is done you can config and prepare your project for production: so basically you use `callBack` option to set the env mode back to `production` so that obsfucation will not run, and then config your app source folder to use `out` folder instead of `src` for production. and the imported tawilwindcss file. if u want css file to be obfuscated to same path just set the ouput directory of postcss directly to `out`
> > 
> > ```
> >  "postcss": "postcss src/**/*.css  --dir out",
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > so basically you use `callBack` option to set the env mode back to `production` and to source folder of production to `out` instead of `src` that way it will work. when you run `npm run build`
> 
> Hi, thanks for your response
> 
> I did the steps you mentioned and I'm almost there, but I have a problem I can't understand why is happening, it looks like a bug
> 
> I run first
> 
> ```shell
>   yarn obfuscate
> ```
> 
> Then
> 
> ```shell
>  yarn build
> ```
> 
> And it's showing some warnings, because I was using classes with arbitrary values before (Like h-[0.5rem]) and the parser was not able to obfuscate them properly, showing those warnings.
> 
> Then I removed the arbitrary values and replaced them for built-in tailwind values, then obfuscated it again and all that. But, when I build it one more time, it still shows warnings (Even when I'm not using the arbitrary values anymore). If I search any of the warning classes, none of them exist! See the photo:
> 
> ![Screenshot_10](https://user-images.githubusercontent.com/47725212/236218267-8a03b2b4-efab-41ce-b91b-1f37e44cffd4.jpg)
> 
> So I ignored warning and renamed out to src and built it, but it was complainig about a syntax error in the CSS. I fixed it and built again, with a successful built, then I opened my project but none of the styles were there. The css classes in the html tags were named different than in the CSS file (For example, in the html class, the name was 'abcd' and in the css file, that class had the 'xyz' name).
> 
> Maybe this last issue is related with warnings thing, I removed css-obfuscator, dist and out folders before building again (I also removed node_modules, and restarted the pc, lol), and the warnings were still there.
> 
> Any hint?
> 
> Thank you!

You welcome ok let me check m but 1st I been updating my answers: also i wld recommend to check this https://stackoverflow.com/questions/66863200/changing-the-input-and-output-directory-in-vite
its better then renaming the `src` and `out` folder as it will cause issues .

**Comment by sort72** [comment]
Actually, I tried the thing in vite.config.js like 1 hour ago, but I was not able to get it working.

I added `root` to my vite config

```js
export default defineConfig({
	plugins: [svgr(), react()],
	root: "src",
});
```

But it said it was not able to find the file `src/index.html` (Which is in root path, not in src), so I was not able to do so. I thought it was a problem more related to vite than to your package, so I was thinking to solve it after I had obfuscation working

**Comment by n4j1Br4ch1D** [comment]
> Actually, I tried the thing in vite.config.js like 1 hour ago, but I was not able to get it working.
> 
> I added `root` to my vite config
> 
> ```js
> export default defineConfig({
> 	plugins: [svgr(), react()],
> 	root: "src",
> });
> ```
> 
> But it said it was not able to find the file `src/index.html` (Which is in root path, not in src), so I was not able to do so. I thought it was a problem more related to vite than to your package, so I was thinking to solve it after I had obfuscation working

@sort72  hi check the replit demo I made this changes:
see if they work for you:

1- install  cross-env
```sh
npm install --save-dev cross-env
```
2. change npm script: set env to `obfuscation` 
```json
 "obfuscate": "cross-env NODE_ENV=obfuscation npm run tailwindcss && npm run postcss"
```
3. I used the `enable` and `callBack` option: 
```js
//postcss.config.cjs 
const isObfscMode = process.env.NODE_ENV === 'obfuscation'; 
module.exports = {
  dir: 'dist/css',
  plugins: [
    require('tailwindcss'),
    require('autoprefixer'),
    require('postcss-obfuscator')({
    /* options */
      enable: isObfscMode, // Force Run on Dev Env or when srcPath equals desPath.
      srcPath: "src", // Source of your files.
      desPath: "out", // Destination for obfuscated html/js/.. files.
      extensions: ['.jsx'],
      formatJson: true, // Format obfuscation data JSON file.
      callBack: function () { // Callback function to call after obfuscation is done.
        process.env.NODE_ENV = "production" // make sure postcss-obfuscator doesnt run  
      } 
    })
  ]
}
```

4. change vite config for `build` command &  `production` env
```js
import { defineConfig } from 'vite'
import react from '@vitejs/plugin-react'

// https://vitejs.dev/config/
export default defineConfig(({ command, mode }) => {
  if (command === 'build' && mode === 'production') {  // build and production specific config
    return {
      base: '',
      root: 'out',
      server: {
        host: '0.0.0.0',
      },
      plugins: [react()],
    }
  } else {
    return {
      server: {
        host: '0.0.0.0',
      },
       plugins: [react()],
    }
  }
})
```

**Comment by sort72** [comment]
I made your suggested changes and I first ran

```shell
yarn obfucaste
```

Then 

```shell
yarn build
```

But I'm facing the following error:

```shell
yarn run v1.22.5
$ vite build
vite v4.2.0 building for production...
✓ 0 modules transformed.
✓ built in 24ms
Could not resolve entry module "out/index.html".
error during build:
RollupError: Could not resolve entry module "out/index.html".
    at error (file:///C:/dev/book/node_modules/rollup/dist/es/shared/node-entry.js:2125:30)
    at ModuleLoader.loadEntryModule (file:///C:/dev/book/booking/node_modules/rollup/dist/es/shared/node-entry.js:23798:20)
    at async Promise.all (index 0)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I saw you changed `root` in vite.config.js to `out` but I don't know why Vite checks if index.html exists there (It's in root directory, outside /out, as a default Vite installation)

I think that this is the last thing we need to fix to get it working

**Comment by n4j1Br4ch1D** [comment]
@sort72  have you tried this?
```json
      "build": "vite build out --emptyOutDir",
```

**Comment by sort72** [comment]
I discovered that we don't need to set a `root` value inside vite config, instead, change the js path in `index.html`

From

```html
<script type="module" src="/src/index.jsx"></script>
````


To

```html
<script type="module" src="/out/index.jsx"></script>
````

How could we handle it with environment vars?

**Comment by sort72** [comment]
So I added 2 .env files: .env.production and .env.dev . And I'm able to switch what folder I want to use on a build (src or out)

`.env.production`
```.env
VITE_OBFUSCATE=true
VITE_SOURCE_PATH='out'
````


`.env.dev`
```.env
VITE_OBFUSCATE=false
VITE_SOURCE_PATH='src'
````

And in `index.html`

```html
<script type="module" src="/%VITE_SOURCE_PATH%/main.jsx"></script>
```

So on production, I need to run

```
yarn obfuscate
yarn build
```

And it creates everything with `out` directory

However, I'm still facing an issue: The css is not shown properly because the classes in html are obfuscated, but the index.css generated file on /dist contains non obfuscated classes (Html: asdhjba, CSS: border-2, for example)

**Comment by n4j1Br4ch1D** [comment]
can i see yout postcss config

**Comment by sort72** [comment]
Here is my postcss config

```js
module.exports = {
	plugins: [
		require("tailwindcss"),
		require("autoprefixer"),
		require("postcss-obfuscator")({
			/* options */
			enable: true, // Force Run on Dev Env or when srcPath equals desPath.
			fresh: true,
			srcPath: "src", // Source of your files.
			desPath: "out", // Destination for obfuscated html/js/.. files.
			extensions: [".jsx", ".js"],
			classPrefix: "msy-", // Prefix for obfuscated CSS class names.
			formatJson: true, // Format obfuscation data JSON file.
			callBack: function () {
				process.env.NODE_ENV = "production"; // make sure postcss-obfuscator doesnt run
			},
		}),
	],
};
```
I'm using `enable: true` for the moment because it seems that process.env.NODE_ENV doesn't get updated with the code you shared on package.json and it's always 'development'. So I'm looking for a solution to this.

I added `fresh: true` to see if I was able to fix something I mentioned before:

I don't understand why the heck this is including a very old css code in the out/index.css (aka input.css) file, as I mentioned earlier today, I had tailwind arbitrary values and it didn't work (You created a new issue about this), so I changed it to tailwind classes to fix and be able to compile the obfuscated css. BUT, I don't understand why it is still adding old css classes with arbitrary values, even when those arbitrary values DON'T exist anywhere on my code and I also reinstalled node_modules...

**Comment by n4j1Br4ch1D** [comment]
I see can you try deleting the `css-obsucator` folder

**Comment by sort72** [comment]
I have removed it so many times without success. Also tried with and without fresh option. I’ll continue trying things and will add an answer with my new findings
________________________________
De: Najib Rachid ***@***.***>
Enviado: Thursday, May 4, 2023 1:51:43 PM
Para: n4j1Br4ch1D/postcss-obfuscator ***@***.***>
Cc: sort72 ***@***.***>; Mention ***@***.***>
Asunto: Re: [n4j1Br4ch1D/postcss-obfuscator] How could I use this within a project using React + Tailwind + Vite? (Issue #2)


I see can you try deleting the css-obsucator folder

**Comment by sort72** [comment]
Thanks for taking the time to handle this issue, the solution you mentioned there seems good for all Tailwind arbitrary values, please do so.

I'll continue my tests related to css names mismatch tomorrow and let you know

**Comment by n4j1Br4ch1D** [comment]
The new Version released, check the release notes:

and to help you with your issue
use this options to debug along with the prefix option:
```
classMethod: 'none'
```
also try excluding some paths or files maybe focus on one file as a start.
```
cssExcludes: Files and paths to exclude from css obfuscation, default is none.
```
or recreate a fresh demo project for testing: see if the same issue will appear.

**Comment by sort72** [comment]
Hi, I'm working on this at the moment but I'm facing some issues with the parser. It seems not to support responsive classes in Tailwind

I created a separated project just for testing this (And uploaded my code [here](https://github.com/sort72/obfuscator-tailwind-test) so you can test)

I just created a new react vite project and added tailwindcss and the obfuscator, and followed the instructions you have given in this topic.

I added a div with 2 classes: bg-red-400 md:bg-blue-400

On dev, if you resize the window to something small, the background is red. And, on bigger screens, the background is blue.

However, on the obfuscated built, the div is always red (It doesn't recognize md:bg-blue-400)

![Screenshot_12](https://user-images.githubusercontent.com/47725212/236835444-6cc4f8cb-714e-4508-b2c8-fe58072f46e4.jpg)

I think the parser is good on the css part, but, the class in the html is malformed. Now it's generated as `msy-md\:bg-blue-400` BUT it should be `msy-md:bg-blue-400`. Removing the `\` fixes that.

![Screenshot_13](https://user-images.githubusercontent.com/47725212/236835806-997fe4cb-ac84-4311-a2e1-2aa54bbd193f.jpg)

That's probably a problem only with prefixes.

**Comment by sort72** [comment]
I tested without the prefix and it doesn't work either:

![Screenshot_14](https://user-images.githubusercontent.com/47725212/236838135-9d6c9ad6-f51f-4871-a950-50152f3539c2.jpg)

The class should be `md:bg-blue-400-obfsctd` instead of `md\:bg-blue-400-obfsctd` (Without the `\`) after the `:` 

(I'm using latest version)

**Comment by n4j1Br4ch1D** [comment]
i just tested the last version as well in the replit , responsive seems working there's still an issue though with arbitrary values replacement in html

**Comment by n4j1Br4ch1D** [comment]
in the replit i have this:
```html
    <main class="pt-[50px] m-4 bg-blue-500 md:bg-red-500">
      <h1 class="text-xl">Welcome to <span class="text-indigo-500">postcss-obsfucator</span></h1>
      <p class="border p-4 m-2 rounded">
        this demo is to try postcss-obsfucator with react, tyescript, vite & tailwindcss
      </p>
    </main>
```
I get this:
```json
{
  ".read-the-docs": ".c-va7vn-c",
  ".m-2": ".c-ecywa-c",
  ".m-4": ".c-fwnil-c",
  ".rounded": ".c-fpb7h-c",
  ".border": ".c-zup9i-c",
  ".bg-blue-500": ".c-alvnn-c",
  ".p-4": ".c-mc2w6-c",
  ".pt-\\[50px\\]": ".c-vi531-c",
  ".text-xl": ".c-qu856-c",
  ".text-indigo-500": ".c-j4e13-c",
  ".md\\:bg-red-500": ".c-ilin8-c"
}
```
```html
    <main class="pt-[50px] c-fwnil-c c-alvnn-c c-ilin8-c">
      <h1 class="c-qu856-c">Welcome to <span class="c-j4e13-c">postcss-obsfucator</span></h1>
      <p class="c-zup9i-c c-mc2w6-c c-ecywa-c c-fpb7h-c">
        this demo is to try postcss-obsfucator with react, tyescript, vite & tailwindcss
      </p>
    </main>
```
```css
.c-ecywa-c {
  margin: 0.5rem;
}

.c-fwnil-c {
  margin: 1rem;
}

.c-fpb7h-c {
  border-radius: 0.25rem;
}

.c-zup9i-c {
  border-width: 1px;
}

.c-alvnn-c {
  --tw-bg-opacity: 1;
  background-color: rgb(59 130 246 / var(--tw-bg-opacity));
}

.c-mc2w6-c {
  padding: 1rem;
}

.c-vi531-c {
  padding-top: 50px;
}

.c-qu856-c {
  font-size: 1.25rem;
  line-height: 1.75rem;
}

.c-j4e13-c {
  --tw-text-opacity: 1;
  color: rgb(99 102 241 / var(--tw-text-opacity));
}

@media (min-width: 768px) {
  .c-ilin8-c {
    --tw-bg-opacity: 1;
    background-color: rgb(239 68 68 / var(--tw-bg-opacity));
  }
}
```

I used prefixes and suffixes. the only issue I see  is `pt-[50px]` doesn't get replaced in HTML , I have re-opend the issue ill fix it soon. I'm I wrong?

**Comment by n4j1Br4ch1D** [comment]
@sort72  ok I reproduced the issue it's related to  ` classMethod: "none"`  used with prefixes and suffixes. ill open an issue and solve both of them soon. thanks so much for your cooperation.  for now u can  comment ` classMethod: "none"`

**Comment by sort72** [comment]
I found another bug. In tailwind, you can use `/` in classes. For example: `md:w-1/2`

That is not mapped properly in the output.css file:

```css
...

.o58kp {
  --tw-shadow: 0 1px 3px 0 rgb(0 0 0 / 0.1), 0 1px 2px -1px rgb(0 0 0 / 0.1);
  --tw-shadow-colored: 0 1px 3px 0 var(--tw-shadow-color), 0 1px 2px -1px var(--tw-shadow-color);
  box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);
}

@media (min-width: 768px) {
  .md\:w-1\/2 {
    width: 50%;
  }

  .od5k0 {
    --tw-bg-opacity: 1;
    background-color: rgb(96 165 250 / var(--tw-bg-opacity));
  }
}
```

css-obfuscator/main.json
```json
{
  ...
  ".shadow": ".o58kp",
  ".md\\:w-1\\\\/2": ".x3bsu",
  ".md\\:bg-blue-400": ".od5k0"
}
```

**Comment by n4j1Br4ch1D** [comment]
> Yeah, with the blank project I created to test it, it seems to be working perfectly. So we can continue with the new issues you created, I'll close this. Thanks!
> 
> Little note: I set `keepData` to false, but it only removed te `css-obfuscator` directory. It would be nice if that property remove the `out` created directory and the `output.css` file as well. Is it possible?

@sort72  glad to hear it open new issues as a feature request.
